velox/exec/WindowPartition.h

```diff
@@ -33,16 +44,41 @@ class WindowPartition {
   /// 'columns' : Input rows of 'data' used for accessing column data from it.
   /// 'sortKeyInfo' : Order by columns used by the the Window operator. Used to
   /// get peer rows from the input partition.
+  /// 'processingUnit' : The processing unit: kPartition or kRow.
   WindowPartition(
       RowContainer* data,
       const folly::Range<char**>& rows,
       const std::vector<exec::RowColumn>& columns,
       const std::vector<std::pair<column_index_t, core::SortOrder>>&
-          sortKeyInfo);
+          sortKeyInfo,
+      ProcessingUnit processingUnit = ProcessingUnit::kPartition);
 
   /// Returns the number of rows in the current WindowPartition.
   vector_size_t numRows() const {
-    return partition_.size();
+    return partition_.size() + offsetInPartition_;
```

Comment:
WindowPartition's can be partial and have only a subset of rows. So just capturing the start and end row ids should be enough for it. The WindowPartition can also have a function to indicate that it is a partial windowPartition.

That would reduce multiple variables here.

Reply:
@aditi-pandit Thanks for your suggestions. I have updated.